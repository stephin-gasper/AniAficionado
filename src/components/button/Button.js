import React from 'react';
import PropTypes from 'prop-types';
import {useTheme} from 'styled-components/native';

import {ButtonContainer, ButtonText, ButtonWrapper} from './Button.style';

const Button = ({
  onPress,
  children,
  iconComponent,
  iconName,
  iconSize,
  style,
  labelStyle,
}) => {
  const theme = useTheme();

  const Icon = iconComponent;

  return (
    <ButtonWrapper>
      <ButtonContainer
        testID="buttonContainer"
        onPress={onPress}
        style={style}
        activeOpacity={0.8}>
        <If condition={iconName}>
          <Icon
            testID="icon"
            name={iconName}
            size={iconSize}
            color={theme.textColor}
          />
        </If>
        <ButtonText testID="buttonText" style={labelStyle}>
          {children}
        </ButtonText>
      </ButtonContainer>
    </ButtonWrapper>
  );
};

Button.propTypes = {
  onPress: PropTypes.func,
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.func,
    PropTypes.elementType,
  ]).isRequired,
  iconComponent: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.func,
    PropTypes.elementType,
  ]),
  iconName: PropTypes.string,
  iconSize: PropTypes.number,
  style: PropTypes.shape({}),
  labelStyle: PropTypes.shape({}),
};

Button.defaultProps = {
  onPress: undefined,
  iconComponent: undefined,
  iconName: undefined,
  iconSize: 15,
  style: {},
  labelStyle: {},
};

export default Button;
