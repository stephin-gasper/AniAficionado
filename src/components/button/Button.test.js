import React from 'react';
import {View as MockView} from 'react-native';
import {render} from 'test/test-utils';

import Button from './Button';

describe('<Button />', () => {
  const sampleText = 'sample';
  it('should render correctly', () => {
    expect(render(<Button>{sampleText}</Button>)).toMatchSnapshot();
  });

  it('should render icon', () => {
    expect(
      render(
        <Button iconComponent={MockView} iconName="someIcon">
          {sampleText}
        </Button>,
      ),
    ).toMatchSnapshot();
  });

  it('should override size of icon when iconSize is present', () => {
    const {getByTestId} = render(
      <Button iconComponent={MockView} iconName="someIcon" iconSize={20}>
        {sampleText}
      </Button>,
    );

    expect(getByTestId('icon').props.size).toBe(20);
  });

  it('should add style to container when style props is present', () => {
    const {getByTestId} = render(
      <Button style={{marginTop: 20}}>{sampleText}</Button>,
    );

    expect(getByTestId('buttonContainer')).toHaveStyleRule('marginTop', 20);
  });

  it('should add style to text when labelStyle props is present', () => {
    const {getByTestId} = render(
      <Button labelStyle={{marginTop: 20}}>{sampleText}</Button>,
    );

    expect(getByTestId('buttonText')).toHaveStyleRule('marginTop', 20);
  });
});
