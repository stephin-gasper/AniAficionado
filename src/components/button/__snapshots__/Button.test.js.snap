// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#352462",
        "borderRadius": 10,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 12,
        "paddingLeft": 25,
        "paddingRight": 25,
        "paddingTop": 12,
      }
    }
    testID="buttonContainer"
  >
    <Text
      style={
        [
          {
            "alignSelf": "center",
            "color": "#fffade",
            "fontSize": 14,
            "fontWeight": "bold",
          },
          {},
        ]
      }
      testID="buttonText"
    >
      sample
    </Text>
  </View>
</View>
`;

exports[`<Button /> should render icon 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#352462",
        "borderRadius": 10,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 12,
        "paddingLeft": 25,
        "paddingRight": 25,
        "paddingTop": 12,
      }
    }
    testID="buttonContainer"
  >
    <View
      color="#fffade"
      name="someIcon"
      size={15}
      testID="icon"
    />
    <Text
      style={
        [
          {
            "alignSelf": "center",
            "color": "#fffade",
            "fontSize": 14,
            "fontWeight": "bold",
          },
          {},
        ]
      }
      testID="buttonText"
    >
      sample
    </Text>
  </View>
</View>
`;
